### PR TITLE
Update doc.go of package downloader

### PR DESCRIPTION
**Pull request by nogang**
## Proposed changes

- Update doc.go of package downloader

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: datasync/downloader/doc.go [hunk 1]
```diff
@@ -18,5 +18,25 @@
...
 // Modified and improved for the klaytn development.
 
-// Package downloader contains the manual full chain synchronisation.
+/*
+Package downloader contains the manual full chain synchronisation.
+
+How downloader works
+
+The downloader is responsible for synchronizing up-to-date status from the peers connected to it. To do this,
+download "headers", "bodies", and "receipts" in parallel, merge them through the pipeline, and reflect them in the state trie.
```
Comment (jeonghonah) [inline datasync/downloader/doc.go]:
```suggestion
it downloads "headers", "bodies", and "receipts" in parallel, merge them through the pipeline, and reflect them in the state trie.
```

File: datasync/downloader/doc.go [hunk 2]
```diff
@@ -18,5 +18,25 @@
...
+
+Source Files
+
+Downloader related functions and variables are defined in the files listed below
```
Comment (KimKyungup) [inline datasync/downloader/doc.go]: Please, put the period at the end of line. :)

File: datasync/downloader/doc.go [hunk 3]
```diff
@@ -18,5 +18,25 @@
...
+  - api.go              : Console api to get synchronization information.
```
Comment (KimKyungup) [inline datasync/downloader/doc.go]: How about changing `api` -> `APIs` in description?

File: datasync/downloader/doc.go [hunk 4]
```diff
@@ -18,5 +18,25 @@
...
+  - downloader.go       : Functions and variables to sync peer and block. And modules for QOS.
+  - downloader_test.go  : Functions for testing the downloader package.
+  - events.go           : Define event type.
+  - metrics.go          : Metric Variables for Packet Transmission and Reception.
```
Comment (KimKyungup) [inline datasync/downloader/doc.go]: `Metric Variables for Packet Transmission and Reception.`
->
`Metric variables for packet transmissions and receptions.`?

File: datasync/downloader/doc.go [hunk 5]
```diff
@@ -18,5 +18,25 @@
...
+The downloader is responsible for synchronizing up-to-date status from the peers connected to it. To do this,
+download "headers", "bodies", and "receipts" in parallel, merge them through the pipeline, and reflect them in the state trie.
+
+Source Files
+
+Downloader related functions and variables are defined in the files listed below
+  - api.go              : Console api to get synchronization information.
+  - downloader.go       : Functions and variables to sync peer and block. And modules for QOS.
+  - downloader_test.go  : Functions for testing the downloader package.
+  - events.go           : Define event type.
```
Comment (KimKyungup) [inline datasync/downloader/doc.go]: `type` -> `types`?

File: datasync/downloader/doc.go [hunk 6]
```diff
@@ -18,5 +18,25 @@
...
+  - metrics.go          : Metric Variables for Packet Transmission and Reception.
+  - modes.go            : Defines the type for SyncMode. SyncMode includes "FullSync", "FastSync", and "LightSync".
+  - peer.go             : Functions that request a packet to the peer and check and set the network status of the peer.
```
Comment (KimKyungup) [inline datasync/downloader/doc.go]: `the peer` -> `a peer`?

File: datasync/downloader/doc.go [hunk 7]
```diff
@@ -18,5 +18,25 @@
...
+
+Source Files
+
+Downloader related functions and variables are defined in the files listed below
+  - api.go              : Console api to get synchronization information.
+  - downloader.go       : Functions and variables to sync peer and block. And modules for QOS.
+  - downloader_test.go  : Functions for testing the downloader package.
+  - events.go           : Define event type.
+  - metrics.go          : Metric Variables for Packet Transmission and Reception.
+  - modes.go            : Defines the type for SyncMode. SyncMode includes "FullSync", "FastSync", and "LightSync".
```
Comment (KimKyungup) [inline datasync/downloader/doc.go]:
```suggestion
  - modes.go            : Definitions of types for SyncMode including "FullSync", "FastSync", and "LightSync".
```

File: datasync/downloader/doc.go [hunk 8]
```diff
@@ -18,5 +18,25 @@
...
+  - peer.go             : Functions that request a packet to the peer and check and set the network status of the peer.
+  - queue.go            : Functions for managing and scheduling received headers, bodies, and receipts.
+  - types.go            : Defines the type of downloaded packet
```
Comment (KimKyungup) [inline datasync/downloader/doc.go]:
```suggestion
  - types.go            : Definitions of the types for downloaded packets.
```

File: datasync/downloader/doc.go [hunk 9]
```diff
@@ -18,5 +18,25 @@
...
+download "headers", "bodies", and "receipts" in parallel, merge them through the pipeline, and reflect them in the state trie.
+
+Source Files
+
+Downloader related functions and variables are defined in the files listed below
+  - api.go              : Console api to get synchronization information.
+  - downloader.go       : Functions and variables to sync peer and block. And modules for QOS.
+  - downloader_test.go  : Functions for testing the downloader package.
+  - events.go           : Define event type.
+  - metrics.go          : Metric variables for packet transmissions and receptions..
```
Comment (nogang) [inline datasync/downloader/doc.go]:
```suggestion
  - metrics.go          : Metric variables for packet transmissions and receptions.
```

File: datasync/downloader/doc.go [hunk 10]
```diff
@@ -18,5 +18,25 @@
...
+Package downloader contains the manual full chain synchronisation.
+
+How downloader works
+
+The downloader is responsible for synchronizing up-to-date status from the peers connected to it. To do this,
+it downloads "headers", "bodies", and "receipts" in parallel, merges them through the pipeline, and reflect them in the state trie.
+
+Source Files
+
+Downloader related functions and variables are defined in the files listed below
```
Comment (nogang) [inline datasync/downloader/doc.go]:
```suggestion
Downloader related functions and variables are defined in the files listed below.
```

File: datasync/downloader/doc.go [hunk 11]
```diff
@@ -18,5 +18,25 @@
...
+  - api.go              : Console api to get synchronization information.
```
Comment (nogang) [inline datasync/downloader/doc.go]:
```suggestion
  - api.go              : Console APIs to get synchronization information.
```

File: datasync/downloader/doc.go [hunk 12]
```diff
@@ -18,5 +18,25 @@
...
+How downloader works
+
+The downloader is responsible for synchronizing up-to-date status from the peers connected to it. To do this,
+it downloads "headers", "bodies", and "receipts" in parallel, merges them through the pipeline, and reflect them in the state trie.
+
+Source Files
+
+Downloader related functions and variables are defined in the files listed below.
+  - api.go              : Console APIs to get synchronization information.
+  - downloader.go       : Functions and variables to sync peer and block. And modules for QOS.
```
Comment (jimni1222) [inline datasync/downloader/doc.go]: What do you think of adding explaination of what QOS is? like '(something)'

File: datasync/downloader/doc.go [hunk 13]
```diff
@@ -18,5 +18,25 @@
...
+  - downloader_test.go  : Functions for testing the downloader package.
+  - events.go           : Definitions of event types.
+  - metrics.go          : Metric variables for packet transmissions and receptions.
+  - modes.go            : A definition of type for SyncMode including "FullSync", "FastSync", and "LightSync".
+  - peer.go             : Functions that request a packet to a peer,  check, and set the network status of a peer.
```
Comment (jimni1222) [inline datasync/downloader/doc.go]:
```suggestion
  - peer.go             : Functions that request a packet to a peer, check, and set the network status of a peer.
```

File: datasync/downloader/doc.go [hunk 14]
```diff
@@ -18,5 +18,25 @@
...
 // Modified and improved for the klaytn development.
 
-// Package downloader contains the manual full chain synchronisation.
+/*
+Package downloader contains the manual full chain synchronisation.
+
+How downloader works
+
+The downloader is responsible for synchronizing up-to-date status from the peers connected to it. To do this,
+it downloads "headers", "bodies", and "receipts" in parallel, merges them through the pipeline, and reflect them in the state trie.
```
Comment (ehnuje) [inline datasync/downloader/doc.go]:
```suggestion
it downloads "headers", "bodies", and "receipts" in parallel, merges them through the pipeline, and reflects them in the state trie.
```

Comment (ehnuje) [reply datasync/downloader/doc.go]: What does it mean "reflect them in the state trie"?

File: datasync/downloader/doc.go (hunk 12 above)
Comment (nogang) [inline datasync/downloader/doc.go]:
```suggestion
  - downloader.go       : Functions and variables to sync peer and block. And modules for QoS (Quality of Service).
```

File: datasync/downloader/doc.go (hunk 14 above)
Comment (nogang) [reply datasync/downloader/doc.go]: This means that the downloaded data("headers", "bodies", and "receipts") will be reflected in the state trie.

**Pull request merged**